contrib/google.golang.org/grpc/option.go

```diff
@@ -52,6 +64,14 @@ func defaults(cfg *config) {
 	cfg.analyticsRate = math.NaN()
 }
 
+// WithServiceName sets the given service name for the intercepted client.
```

Comment:
since this is a grpc middleware...is there value in having folks pass in `f` vs just enabling the dynamic grpc-service-name function?  Seems like lots of folks might want the dynamic parsing thing in just the way we're thinking

Reply:
Yeah, I was worried about backwards compatibility though, for folks who are already using WithServiceName(str).  I would like to upstream this PR though, once we have a good handle on it working for us.

Reply:
oh yeah, sorry this comment was about your WithDynamicServiceName func (which this comment is on) -- thinking it could not even take any arguments and within itself setup the use of the function you'd intended to pass in

Reply:
Ah, yeah.  I was imaging that someone might manipulate the info in not quite the same way we do. Like maybe they want the package prefix too.